File: ruby/rna-transcription/rna_transcription.rb
class Complement
  def self.of_dna(input)
    strands = {'G' => 'C', 
               'C' => 'G', 
               'T' => 'A', 
               'A' => 'U'}
    string = ''
    input.split('').map do |letter|
      string += strands[letter]
    end
    return string
  end
end

File: ecmascript/rna-transcription/rna-transcription.js
class Transcriptor {
  toRna(str) {
    const rnaArr = str.split('').map(letter => {
      const rnaLetter = rnaMap[letter]
      if(rnaLetter){
        return rnaLetter
      }else {
        throw new Error("Invalid input DNA")
      }
    })
    return rnaArr.join('')
  }
}

const rnaMap = {
  'G': 'C',
  'C': 'G',
  'T': 'A',
  'A': 'U'
}

export default Transcriptor